cuda/solver/common_trs_kernels.cuh

```diff
@@ -577,7 +589,13 @@ void sptrsv_naive_caching(std::shared_ptr<const CudaExecutor> exec,
                 as_device_type(x->get_values()), x->get_stride(), n, nrhs,
                 unit_diag, nan_produced.get_data(), atomic_counter.get_data());
     } else {
-        sptrsv_naive_caching_kernel<is_upper>
+        // TODO: need to investigate
+        // memory operation on the half-precision shared_memory seem to give
+        // wrong result. we use float in shared_memory.
+        using shared_value_type = std::conditional_t<
+            std::is_same<remove_complex<ValueType>, gko::half>::value, float,
+            ValueType>;
+        sptrsv_naive_caching_kernel<is_upper, device_type<shared_value_type>>
```

Comment:
now this will also be `float` when using `std::complex<gko::half>`. That doesn't seem correct. You will loose any imaginary part that might be in the matrix or vectors.

Reply:
you are right.
Sorry, I do not pay enough attention when changing it.